src/main/java/core/basesyntax/MainApp.java

```diff
@@ -2,4 +2,13 @@
 
 public class MainApp {
 
+    public static void mail (String[] args) {
+        Machine [] machines = new Machine[] {
+                new Truck(), new Bulldozer(), new Excavator()
+        };
+        for (int i = 0; i <= machines.length; i++) {
```

Comment:
The loop condition `i <= machines.length` should be `i < machines.length` to prevent an `ArrayIndexOutOfBoundsException`. The current condition will attempt to access an index equal to `machines.length`, which is out of bounds.